File: solutions/3107. Minimum Operations to Make Median of Array Equal to K/3107.py
# Time complexity: O(sort)
# Space complexity: O(sort)
class Solution:
  def minOperationsToMakeMedianK(self, nums: List[int], k: int) -> int:
    ans = 0

    nums.sort()
    for i in range(len(nums) // 2 + 1):
      ans += max(0, nums[i] - k)
    for i in range(len(nums) // 2, len(nums)):
      ans += max(0, k - nums[i])

    return ans

File: solutions/3107. Minimum Operations to Make Median of Array Equal to K/3107.cpp
// Time complexity: O(sort)
// Space complexity: O(sort)
class Solution {
 public:
  long long minOperationsToMakeMedianK(vector<int>& nums, int k) {
    long long ans = 0;

    ranges::sort(nums);
    for (int i = 0; i <= nums.size() / 2; ++i)
      ans += max(0, nums[i] - k);
    for (int i = nums.size() / 2; i < nums.size(); ++i)
      ans += max(0, k - nums[i]);

    return ans;        
  }
};